### PR TITLE
CORE-1095 HTTP server changes

### PR DESCRIPTION
**Pull request by rabbitonweb**
## Overview
In this PR we:

[x] Remove dedicated metrics server (we keep /metrics endpoint on http server)
[x] Remove metrics-port from configuration
[x] Remove Lykke and JSON api

### Which JIRA issue does this PR relate to? If there is not a JIRA issue addressing this work, please create one now and add the link here.
CORE-1095

### CODE REVIEW

**Comment by dzajkowski** [comment]
I think that @adaszko and @woky should be involved in this PR.

**Comment by adaszko** [comment]
Educate me, please: This PR makes it so that the only remaining and canonical source of rnode metrics is the [HTTP endpoint that gets polled by Prometheus](https://github.com/adaszko/rchain/blob/b75c0118f0ac71b6364eb69cd49f0e788dac55c2/node/src/main/scala/coop/rchain/node/diagnostics/NewPrometheusReporter.scala#L91-L93) ?

**Comment by rabbitonweb** [comment]
@adaszko 
we used to have:
1. an http server that did nothing really (hosted /lykke endpoint - but that was a "mock" implementation")
2. Prometheus reporter that had his own internal, embedded server to host report to Prometheus

Now, that embedded server was causing issues. Was slow and was throwing errors.

I've created a new Prometheus reporter, that was an almost "copy-paste" from the original one - the only difference was that I've removed the embedded server from it. What I did was created an new endpoint (/metrics) on existing http server. 

However my changes were done 1 week before launching of a test net, we thought it will be good to keep somewhat "backward" compatibility. That's why I've launched a *second* http server that would also host the metrics endpoint so that it would act as the embedded server (that was just removed).

In other words: for the past two weeks we were running two http servers: one holding lykke & metrics (behind  --port) and second (obsolete one) holding only metrics (--metrics-port).

Aftert this PR:
1. --metrics-port will be removed
2. second http server will be removed
3. first http server will only host metricies (because lykke feature are not needed at the momemtn)

File: node/src/main/scala/coop/rchain/node/configuration/Configuration.scala [hunk 1]
```diff
@@ -36,9 +36,8 @@ object Configuration {
 
   private val DefaultPort                       = 40400
   private val DefaultGrpcPortExternal           = 40401
-  private val DefaultHttPort                    = 40402
-  private val DefaultMetricsPort                = 40403
-  private val DefaultGrpcPortInternal           = 40404
+  private val DefaultGrpcPortInternal           = 40402
+  private val DefaultHttPort                    = 40403
```
Comment (sebtomba) [inline node/src/main/scala/coop/rchain/node/configuration/Configuration.scala]: @rabbitonweb Please run integration tests on your local machine before merging to `dev`.

Comment (rabbitonweb) [reply node/src/main/scala/coop/rchain/node/configuration/Configuration.scala]:
```
===========================================================
=================TEST SUMMARY RESULTS======================
PASS: peer1.rchain.coop: Peers count correct in node logs.
PASS: peer0.rchain.coop: Peers count correct in node logs.
PASS: peer1.rchain.coop: Rholang evaluation of files performed correctly.
PASS: peer0.rchain.coop: Rholang evaluation of files performed correctly.
PASS: peer0.rchain.coop: REPL loader success!
PASS: peer1.rchain.coop: Proposal of blocks for deployed contracts worked.
PASS: peer0.rchain.coop: Proposal of blocks for deployed contracts worked.
PASS: bootstrap.rchain.coop: Proposal of blocks for deployed contracts worked.
PASS: peer1.rchain.coop: No errors defined by "ERROR" in logs.
PASS: peer0.rchain.coop: No errors defined by "ERROR" in logs.
PASS: peer1.rchain.coop: No text of "RuntimeException" in logs.
PASS: peer0.rchain.coop: No text of "RuntimeException" in logs.
PASS ALL: All tests successfully passed
===========================================================
===========================================================
```
